integrations/osmosis-adapter/src/dex.rs

```diff
@@ -7,7 +7,7 @@ use crate::{AVAILABLE_CHAINS, OSMOSIS};
 #[derive(Default)]
 pub struct Osmosis {
     pub version_control_contract: Option<VersionControlContract>,
-    pub local_proxy_addr: Option<Addr>,
+    pub sender: Option<Addr>,
```

Comment:
sender? 

I think this makes the logic confusing, same with having sender in fetch_data function. 

`sender` doesn't matter for adapters. What matters is which address the messages should be constructed for. So maybe we can rename it to `construct_for_addr` or `addr_as_sender` or similar but `sender` itself implies that it is the actual caller of this message, which it almost never is.

Reply:
Agree, this will be a sender of the generated message! It's hard to get not confusing name for this... `action_sender`?

Reply:
renamed it to `addr_as_sender` @CyberHoward